plugins/woocommerce/tests/php/src/Internal/DataStores/Fulfillments/FulfillmentsDataStoreHookTest.php

```diff
@@ -131,6 +131,103 @@ function ( $fulfillment ) use ( &$hook_called, &$received_fulfillment ) {
 		$this->assertCount( 1, $fulfillments, 'Fulfillment was not created.' );
 	}
 
+	/**
+	 * Test that the fulfillment before update hook is called when updating a fulfillment.
+	 */
+	public function test_fulfillment_before_update_hook_is_called() {
+		// Create a fulfillment for the order.
+		$fulfillment = $this->get_test_fulfillment( $this->order->get_id() );
+		$fulfillment->save();
+		$this->assertNotNull( $fulfillment->get_id(), 'Fulfillment ID should not be null.' );
+
+		$hook_called = false;
+		add_filter(
+			'wc_fulfillment_before_update',
+			function ( $fulfillment ) use ( &$hook_called ) {
+				$hook_called = true;
+				return $fulfillment;
+			}
+		);
+
+		// Add a modification to the saved fulfillment, so we can see the difference.
+		$fulfillment->add_meta_data( 'test_meta_update', 'test_meta_value' );
+
+		$this->store->update( $fulfillment );
+		$this->assertTrue( $hook_called, 'The fulfillment before update hook was not called.' );
+
+		$db_fulfillment = new Fulfillment( $fulfillment->get_id() );
+		$this->assertTrue( $db_fulfillment->meta_exists( 'test_meta_update' ), 'Fulfillment was not updated.' );
+	}
+
+	/**
+	 * Test that the fulfillment before update hook can prevent updating a fulfillment.
+	 */
+	public function test_fulfillment_before_update_hook_can_interrupt() {
+		// Create a fulfillment for the order.
+		$fulfillment = $this->get_test_fulfillment( $this->order->get_id() );
+		$fulfillment->save();
+		$this->assertNotNull( $fulfillment->get_id(), 'Fulfillment ID should not be null.' );
+
+		$hook_called = false;
+		add_filter(
+			'wc_fulfillment_before_update',
+			function () use ( &$hook_called ) {
+				$hook_called = true;
+				throw new \Exception( 'Fulfillment update prevented by hook.' );
```

Comment:
`\FulfillmentException` maybe?

Reply:
Nope, that's only for rendering the error on the fulfillment panel. Any exception can cancel the action.